yal/optout.py

```diff
@@ -177,33 +177,27 @@ async def state_delete_saved(self):
         msisdn = utils.normalise_phonenumber(self.inbound.from_addr)
         whatsapp_id = msisdn.lstrip(" + ")
 
-        data = {
-            "onboarding_completed": "",
-            "opted_out": "TRUE",
+        rp_fields = await rapidpro.get_instance_fields()
+        rp_field_keys = [rp_field["key"] for rp_field in rp_fields]
+        rp_field_keys_set = set(rp_field_keys)
```

Comment:
I like the python list comprehension here.

You could do this in one line:
`rp_field_keys = set([rp_field["key"] for rp_field in rp_fields])`

Then just use `rp_field_keys` going forward

Reply:
👍